tools/devops/automation/build-pipeline.yml

```diff
@@ -84,25 +84,39 @@ parameters:
       stageName: 'mac_10_15',
       displayName: 'Mac Catalina (10.15)',
       macPool: 'VSEng-Xamarin-RedmondMacBuildPool-iOS-Trusted',
-      osVersion: '10.15',
       statusContext: 'Mac Catalina (10.15)',
-      checkDemands: true
+      demands: [
+        "Agent.OS -equals Darwin",
+        "Agent.OSVersion -equals '10.15'"
+      ]
     },
     {
       stageName: 'mac_10_14',
       displayName: 'Mac Mojave (10.14)',
       macPool: 'Hosted Mac Internal Mojave',
-      osVersion: '10.14',
       statusContext: 'Mac Mojave (10.14)',
-      checkDemands: false
+      demands: [
+        "Agent.OS -equals Darwin",
```

Comment:
does not it need something like `"Agent.OSVersion -equals '10.14'"` ? 
or is the `macPool` enough to identify them ?

Reply:
In this case, we do not need to do it because the agent did not set it, meaning that if we add it we never get a match (bad configured agent). But the pools only contain machines of that version.